### DISCUSSION TITLE
Interface kernel advice needed

### DISCUSSION THREAD

**Discussion by joe61vette** [idaholab/moose#23842] [Q&A General] [answered]
Hello:

In one block I have a temperature variable that is "constant monomial".  In an adjacent block, a different temperature variable is "first Lagrange".  Both are FEM, so I am not mixing FEM and FVM.

I would like to place something like an interface kernel between the two blocks to allow conduction heat transfer across the boundary.  Is there an existing interface kernel for such a case?  Is there an example somewhere for something like this?

Thanks,
  Joe Kelly

**Comment by jiangwen84** [comment]
> Hello:
> 
> 
> 
> In one block I have a temperature variable that is "constant monomial".  In an adjacent block, a different temperature variable is "first Lagrange".  Both are FEM, so I am not mixing FEM and FVM.
> 
> 
> 
> I would like to place something like an interface kernel between the two blocks to allow conduction heat transfer across the boundary.  Is there an existing interface kernel for such a case?  Is there an example somewhere for something like this?
> 
> 
> 
> Thanks,
> 
>   Joe Kelly

Depends on your problems, you can use either thermal contact https://mooseframework.inl.gov/modules/combined/tutorials/introduction/step01.html or thin layer heat transfer https://mooseframework.inl.gov/moose/source/interfacekernels/ThinLayerHeatTransfer.html

**Reply by joe61vette** [reply]
Thanks Wen.  Do you know if either of these work when the variables are of different types?  I basically have DG on one side and CFEM on the other side.  I am about to quit for today and will give this a try tomorrow.

**Reply by jiangwen84** [reply]
> Thanks Wen.  Do you know if either of these work when the variables are of different types?  I basically have DG on one side and CFEM on the other side.  I am about to quit for today and will give this a try tomorrow.

Theroetically, I do not see any issues with coupling different types of elements.  However, I believe this has not been tested before,  so you might have some troubles there.  Let us know what you find out.

**Comment by joe61vette** [comment]
I have tried three approaches without success.  The input files are attached.
[interface.zip](https://github.com/idaholab/moose/files/11072010/interface.zip)

InterfaceTester.i : uses an internal sideset between the 2 blocks. Mesh is not split.  SideSetHeatTransferKernel is used. Result is non-convergence due to a ZEROPIVOT.

InterfaceTester_2.i :  Mesh is split.  ThinLayerHeatTransfer is used. Result is a segfault.

InterfaceTester_3.i : uses an internal sideset between the 2 blocks. Mesh is not split.  InterfaceDiffusion is used. Result is non-convergence due to a ZEROPIVOT.

Any ideas?

Thanks.

**Reply by GiudGiud** [reply, marked as answer]
Tying this to the other thread: https://github.com/idaholab/moose/discussions/23846

Seg fault in test case 2 is definitely a problem. We'll fix it
For 1 and 3, please see the other thread.
